File: server/src/engine/geometry/vector2.ts
class Vector2 {

    constructor(public x: number, public y: number) {
    }

    combine(other: Vector2): Vector2{
        return new Vector2(this.x + other.x, this.y + other.y);
    }

    
    add(other: Vector2) {
        var dx = other.x - this.x;
        var dy = other.y - this.y;
        return Math.abs(Math.sqrt((dx * dx) + (dy * dy)));
    }
    
    distance(pos) {
        var dx = pos.x - this.x;
        var dy = pos.y - this.y;
        return Math.abs(Math.sqrt((dx * dx) + (dy * dy)));
    }

    manhattan(pos) {
        return (Math.abs(this.x - pos.x) + Math.abs(this.y - pos.y));
    }

    clone() {
        return (new Vector2(this.x, this.y));
    }

    toString() {
        return ("(" + this.x + ", " + this.y + ")");
    }
}

export = Vector2;

File: sharpServer/Engine/Geometry/Vector2.cs
using System;

namespace SharpServer.Engine
{
    public class Vector2
    {
        public int X;
        public int Y;

        public Vector2(int x, int y)
        {
            X = x;
            Y = y;
        }

        public Vector2 Clone()
        {
            return new Vector2(X, Y);
        }

        public Vector2 Combine(Vector2 other)
        {
            return new Vector2(X + other.X, Y + other.Y);
        }

        public double Distance(Vector2 other)
        {
            var dx = other.X - X;
            var dy = other.Y - Y;
            return Math.Sqrt((dx*dx) + (dy*dy));
        }

        public int Manhattan(Vector2 other)
        {
            return Math.Abs(X - other.X) + Math.Abs(Y - other.Y);
        }

        public override string ToString()
        {
            return "(" + X + ", " + Y + ")";
        }
    }
}